cmd/kubeadm/app/cmd/join.go

```diff
@@ -61,28 +61,66 @@ func NewCmdJoin(out io.Writer) *cobra.Command {
 	var cfgPath string
```

Comment:
The defaulting is broken right now. Now it defaults an empty NodeConfiguration object, while we expect it to default a NodeConfiguration object with the flag values present. We'll solve this by doing the following above instead of the four lines currently:
```go
	cfg := &kubeadmapiext.NodeConfiguration{}
	api.Scheme.Default(cfg)
```
This runs the initial defaulting of the external (versioned) config object, so defaults for an empty object will show up in the cobra help text